src/module.c

```diff
@@ -1064,12 +1063,16 @@ int RM_CreateSubcommand(RedisModuleCommand *parent, const char *name, RedisModul
         return REDISMODULE_ERR; /* A parent command should be a pure container of subcommands */
 
     /* Check if the command name is busy within the parent command. */
-    if (parent_cmd->subcommands_dict && lookupCommandByCStringLogic(parent_cmd->subcommands_dict, name) != NULL)
+    if (parent_cmd->subcommands_dict && lookupSubCommandByCStringLogic(parent_cmd->subcommands_dict, parent_cmd->fullname, name) != NULL)
         return REDISMODULE_ERR;
 
     RedisModuleCommand *cp = moduleCreateCommandProxy(parent->module, name, cmdfunc, flags, firstkey, lastkey, keystep);
     cp->rediscmd->arity = -2;
 
+    sds fullname = sdscatfmt(sdsempty(), "%s|%s", parent_cmd->fullname, cp->rediscmd->fullname);
```

Comment:
i think we should have a function that just returns `sdscatfmt(sdsempty(), "%s|%s", parent_name, sub_name)`